### PR TITLE
Fix transit tube atmos (again)

### PR DESCRIPTION
**Pull request by pubby**
#30116 had the right idea but it did not fix the problem. Calling `archive` on the turf's gas mixture was the best fix I've found.

I'm not sure how safe it is to call `archive` like this. If there's concern, I guess it would be possible to clone the turf's gas archive and then copy it back after `share` returns. I'm not sure if that would be better or worse.

I've also added `air_update_turf` so that the released gas gets mixed into the surrounding atmosphere correctly. Otherwise the gas changes remain invisible until something else makes the turf active.

### CODE REVIEW

**Comment by pubby** [comment]
Kinda unrelated but I wonder if `atmos_adjacent_turfs` should be changed to a reciprocal to avoid the division in `share`.

**Comment by YPOQ** [comment]
Good PR. I'll try to avoid touching atmos again until I understand it a bit better.

**Comment by pubby** [comment]
me too buddy

File: code/game/objects/structures/transit_tubes/transit_tube_pod.dm [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 /obj/structure/transit_tube_pod/New(loc)
 	..()
-	air_contents.assert_gases("o2", "n2")
+	air_contents.add_gases("o2", "n2")
```
Comment (Cyberboss) [inline code/game/objects/structures/transit_tubes/transit_tube_pod.dm]: Why this change, also port to Initialize please

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/transit_tubes/transit_tube_pod.dm [hunk 2]
```diff
@@ -9,9 +9,9 @@
 	var/datum/gas_mixture/air_contents = new()
 
 
-/obj/structure/transit_tube_pod/New(loc)
-	..()
-	air_contents.assert_gases("o2", "n2")
+/obj/structure/transit_tube_pod/Initialize()
+	. = ..()
+	air_contents.add_gases("o2", "n2")
```
Comment (Cyberboss) [inline code/game/objects/structures/transit_tubes/transit_tube_pod.dm]: still wondering why you changed it from assert_gases?

File: code/game/objects/structures/transit_tubes/transit_tube_pod.dm (hunk 1 above)
Comment (pubby) [reply code/game/objects/structures/transit_tubes/transit_tube_pod.dm]: `assert_gases` is just `add_gases` with a condition to return early if the gas already exists.

But we know the gas does not exist yet, so using `add_gases` here is slightly more idiomatic and also a tiny micro-optimization.

File: code/game/objects/structures/transit_tubes/transit_tube_pod.dm (hunk 2 above)
Comment (pubby) [reply code/game/objects/structures/transit_tubes/transit_tube_pod.dm]: see comment above

**Review by Cyberboss — APPROVED** [review]

File: code/game/objects/structures/transit_tubes/station.dm [hunk 3]
```diff
@@ -147,8 +147,11 @@
 		sleep(OPEN_DURATION + 2)
 		pod_moving = 0
 		if(!QDELETED(pod))
+			var/datum/gas_mixture/floor_mixture = loc.return_air()
+			floor_mixture.archive()
```
Comment (Cyberboss) [inline code/game/objects/structures/transit_tubes/station.dm]: Is it possible for floor_mixture to be null?

**Comment by PraiseRatvar** [comment]
i hope you tested this

File: code/game/objects/structures/transit_tubes/station.dm (hunk 3 above)
Comment (pubby) [reply code/game/objects/structures/transit_tubes/station.dm]: I don't *think* so, and the previous code didn't check for it either.

Want me to add a check for peace of mind?

Comment (ShizCalev) [reply code/game/objects/structures/transit_tubes/station.dm]: return_air() can only return as null if it's not called on a loc, otherwise there is always a gas mixture present on a tile.

**Comment by Cyberboss** [comment]
I keep forgetting to merge this

**Comment by duncathan** [comment]
curious little fix not bad
